### DISCUSSION TITLE
Wireless Mesh Networking for IoT wireless platforms; BTLE & LoRA  (FruityMesh/Meshtastic)

### DISCUSSION THREAD

**Discussion by TheRook** [libp2p/rust-libp2p#4526] [Ideas]
## Description

Provide better layer1 wireless mesh networking support to better intigrate Bluetooth Low Energy 'btle' and Long Range public transmission over LoRa and LoRaWAN. 

## Motivation

An embedded device, or collection of IoT devices could use libp2p to communicate with other devices in the vicinity or connect to a device with internet access to access the global libp2p network. 

## Requirements

1. This is an extension of an existing effort:
https://github.com/libp2p/rust-libp2p/issues/167

2. Make sure that bluetooth and LoRa meshing are apart of the future roadmap.

3. Study FruityMesh, Meshtastic and other opens source btle and LoRA communication platforms to see how they can improve mobile, browser-based, and embedded communication with libp2p.

## Open questions

What does the community think?

## Are you planning to do it yourself in a pull request?

Maybe

**Comment by umgefahren** [comment]
Isn't there already an effort for Bluetooth? #971 As far as I know we are currently limited on the technical site.
